tokio-util/src/codec/mod.rs

```diff
@@ -7,6 +7,61 @@
 //! [`AsyncWrite`], to framed streams implementing [`Sink`] and [`Stream`].
 //! Framed streams are also known as transports.
 //!
+//! ## Example using codecs
+//! ```
+//! use tokio_util::codec::LinesCodec;
+//!
+//! use futures::sink::SinkExt;
+//! use tokio_util::codec::FramedWrite;
+//!
+//! use tokio_stream::StreamExt;
+//! use tokio_util::codec::FramedRead;
+//!
+//! #[tokio::main]
+//! async fn main() {
+//!     let messages = vec!["Hello", "World"];
+//!     let buffer = Vec::new();
+//!
+//!     // Encoding
+//!     let encoder = LinesCodec::new();
+//!
+//!     // FramedWrite is a sink which means you can send values into it
+//!     // asynchronously. Other examples of a sink are channels and socekts.
+//!     // Values sent into a FramedWrite will be framed according to a codec.
+//!     let mut writer = FramedWrite::new(buffer, encoder);
+//!
+//!     for message in &messages {
+//!         // To be able to send values into a FramedWrite, you need to bring the
+//!         // `futures::sink::Sink` into scope. But working with `Sink` is difficult
+//!         // since it only provides the barebone functions. In order to work with a `Sink`
+//!         // in a higher level, you should use the functions in the `SinkExt` trait.
```

Comment:
I would just say that you need `SinkExt` in scope. It's more confusing to say "you need `Sink` in scope, actually you don't need `Sink` in scope you need `SinkExt` instead".